### PR TITLE
Test Cases for Review

### PR DESCRIPTION
**Pull request by Cotto4**
Judge away!

### CODE REVIEW

File: QuestionApp/src/ca/ualberta/cs/cmput301f14t14/questionapp/DataManager.java [hunk 1]
```diff
@@ -51,4 +54,21 @@ public void enableNetworkAccess() {
 		
 	}
 
+	public List<Object> queryKeywords(String string) {
```
Comment (stephenjust) [inline QuestionApp/src/ca/ualberta/cs/cmput301f14t14/questionapp/DataManager.java]: Ooooh, object?? Try "? extends Model" at least.

File: QuestionAppTests/src/ca/ualberta/cs/cmput301f14t14/questionapp/test/ImageTest.java [hunk 2]
```diff
@@ -20,17 +20,20 @@ protected void tearDown() throws Exception {
 	}
 	
 	/**
-	 * Test UC2 - Attach an Image
+	 * UC7 TC7.1, 7.2 - Attach an Image
 	 */
 	
 	public void testAddImage() {
-		Image image = new Image(null, null);
+		Image image = new Image(Uri.parse("~\\QuestionAppTests\\res\\drawable-hdpi\\under64.jpg"), null);
```
Comment (stephenjust) [inline QuestionAppTests/src/ca/ualberta/cs/cmput301f14t14/questionapp/test/ImageTest.java]: This won't work. Drawable.fromBitmap(?) (R.drawable.under64).getUri() or something like that should work.

File: QuestionAppTests/src/ca/ualberta/cs/cmput301f14t14/questionapp/test/SearchTest.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+	protected void tearDown() throws Exception {
+		super.tearDown();
+	}
+	
+	/**
+	 * UC10 TC10.1 - Empty Search
+	 */
+	
+	public void testEmptySearch() {
+		List<Object> results = manager.queryKeywords("1-9xkjshdiused999999999fihskdfjfsfkj");
```
Comment (stephenjust) [inline QuestionAppTests/src/ca/ualberta/cs/cmput301f14t14/questionapp/test/SearchTest.java]: Eww!

File: QuestionAppTests/src/ca/ualberta/cs/cmput301f14t14/questionapp/test/SearchTest.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+	}
+	
+	/**
+	 * UC10 TC10.2 - Offline Search
+	 */
+	
+	public void testOfflineSearch() {
+		try {
+		manager.disableNetworkAccess();
+		List<Object> results = manager.queryKeywords("");
```
Comment (stephenjust) [inline QuestionAppTests/src/ca/ualberta/cs/cmput301f14t14/questionapp/test/SearchTest.java]: Vile

File: QuestionAppTests/src/ca/ualberta/cs/cmput301f14t14/questionapp/test/SearchTest.java [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+			//Passed
+		}
+	}
+	
+	/**
+	 * UC10 TC10.3 - Proper Search
+	 */
+	
+	public void testRealSearch() {
+		List<Object> results = manager.queryKeywords("jeans");
```
Comment (stephenjust) [inline QuestionAppTests/src/ca/ualberta/cs/cmput301f14t14/questionapp/test/SearchTest.java]: Preposterous

**Comment by stephenjust** [comment]
Also, merge conflicts! You can fix those with "git merge upstream master"

**Comment by Cotto4** [comment]
What do you suggest for the Search test cases? I had no ideas so I wrote some crap down.

**Comment by stephenjust** [comment]
The right way would be to have some kind of MockSearchProvider I think.
Then it can return exactly what we want so the results are consistent.
Basically the tests are "Don't break when there's no results" and "Don't
break when there are results"

## 

Stephen Just

**Comment by Cotto4** [comment]
So something like 

class MockSearchProvider

private List<? extends Model> allItems

public MockSearchProvider() {
allItems.add(new Question("","",null);
allItems.add(new Answer("",null);
}

public List<? extends Model> query(String keywords) {
//comma separate keywords
List<? extends Model> results = new ArrayList
foreach item in allItems{
 foreach word in keywords{

if word in item.title or word in item.body
 results.add(item);
}

return results;

Just add some fake data into the constructor of a new class, and query that data with our search? I just don't think that is testing anything that we would actually implement, I think that I am misunderstanding.

**Comment by stephenjust** [comment]
It isn't that useful a test without views. If we have to test sorting
search items for example, then it could be useful. Or testing that our
search view distinguishes between questions and answers.

**Comment by Cotto4** [comment]
So, since we don't have views, I'm still stuck on how to write any sort of a test for search.

**Comment by Cotto4** [comment]
I wrote some stuff, but it doesn't really fix the problem at all. You seem to understand this much better than I do, think you can fine tune it?

**Comment by stephenjust** [comment]
Yup

## 

Stephen Just

**Comment by Cotto4** [comment]
Sweet thanks man. And apparently git merge upstream master isn't a thing...

merge: upstream - not something we can merge

Any ideas?

**Comment by stephenjust** [comment]
I erred. It's git pull, not git merge.

**Comment by Cotto4** [comment]
Alright did that, let me know if I can do anything else for ya!

**Comment by stephenjust** [comment]
Approved. Once this is merged I'll take care of the leftovers.

**Pull request merged**